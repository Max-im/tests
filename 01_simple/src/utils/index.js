export const getNexId = arr => {
  let maxId = 0;
  arr.forEach(item => {
    if (maxId < item.id) maxId = item.id;
  });
  return [{ id: maxId + 1 }, ...arr];
};
