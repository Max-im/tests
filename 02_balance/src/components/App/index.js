import React from "react";
import Wallet from "../Wallet";
import Loot from "../Loot";

export default function index() {
  return (
    <div>
      <h2>Balance</h2>
      <hr />
      <Wallet data-test="app-wallet" />
      <hr />
      <Loot data-test="app-loot" />
    </div>
  );
}
