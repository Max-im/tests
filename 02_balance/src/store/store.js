import { createStore } from "redux";
import rootReducer from "./reducers";
import { applyMiddleware } from "redux";
import thunk from "redux-thunk";

const initialstate = {};

export default createStore(rootReducer, initialstate, applyMiddleware(thunk));
