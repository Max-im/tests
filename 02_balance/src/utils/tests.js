import React from "react";
import renderer from "react-test-renderer";

export const isRender = component => {
  const c = renderer.create(component);
  expect(c.toJSON()).toMatchSnapshot();
};
