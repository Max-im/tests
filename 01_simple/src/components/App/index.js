import React, { Component } from "react";
import { Button } from "react-bootstrap";
import { getNexId } from "../../utils";
import Gifts from "../Gifts";

class App extends Component {
  state = {
    gifts: []
  };

  addGift = () => {
    this.setState({ gifts: getNexId(this.state.gifts) });
  };

  removeGift = id => {
    const gifts = this.state.gifts.filter(item => item.id !== id);
    this.setState({ gifts });
  };

  render() {
    return (
      <div className="app">
        <h2>Gift Giver</h2>
        <div className="gift-list">
          {this.state.gifts.map(gift => (
            <Gifts key={gift.id} gift={gift} removeGift={this.removeGift} />
          ))}
        </div>
        <Button className="btn-add" onClick={this.addGift}>
          Add Gift
        </Button>
      </div>
    );
  }
}

export default App;
