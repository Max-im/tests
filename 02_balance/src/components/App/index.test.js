import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import { isRender } from "../../utils/tests";
import store from "../../store/store";
import App from "./";
Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
  const app = shallow(<App />);
  test("render successful", () =>
    isRender(
      <Provider store={store}>
        <App />
      </Provider>
    ));

  test("Wallet exists", () => {
    expect(app.find("[data-test='app-wallet']").exists()).toBe(true);
  });

  test("Loot exists", () => {
    expect(app.find("[data-test='app-loot']").exists()).toBe(true);
  });
});
