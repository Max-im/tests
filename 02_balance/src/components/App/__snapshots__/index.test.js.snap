// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render successful 1`] = `
<div>
  <h2>
    Balance
  </h2>
  <hr />
  <div>
    <h3
      className="balance"
    >
      Balance 
      0
    </h3>
    <br />
    <input
      className="input-wallet"
      onChange={[Function]}
    />
    <button
      className="btn-deposit"
      onClick={[Function]}
    >
      add deposit
    </button>
    <button
      className="btn-withdraw"
      onClick={[Function]}
    >
      withdraw
    </button>
  </div>
  <hr />
  <div>
    <h3>
      Bitcoin balance: 
      
    </h3>
  </div>
</div>
`;
